x-pack/legacy/plugins/integrations_manager/server/plugin.ts

```diff
@@ -33,21 +33,26 @@ export class Plugin {
   public setup(core: CoreSetup) {
     const { http, elasticsearch } = core;
     const { server } = http;
-    const context: PluginContext = {
+    const pluginContext: PluginContext = {
       esClient: elasticsearch.createClient(PLUGIN_ID),
     };
 
-    // map routes to handlers
-    routes.forEach(route => {
-      // make these items available to handlers via h.context
-      // https://github.com/hapijs/hapi/blob/master/API.md#route.options.bind
-      // aligns closely with approach proposed in handler RFC
-      // https://github.com/epixa/kibana/blob/rfc-handlers/rfcs/text/0003_handler_interface.md
+    // make pluginContext entries available to handlers via h.context
+    // https://github.com/hapijs/hapi/blob/master/API.md#route.options.bind
+    // aligns closely with approach proposed in handler RFC
+    // https://github.com/epixa/kibana/blob/rfc-handlers/rfcs/text/0003_handler_interface.md
+    const routesWithContext = routes.map(function injectRouteContext(route) {
+      // merge route.options.bind, defined or otherwise, into pluginContext
+      // routes can add extra values or override pluginContext values (e.g. spies, etc)
       if (!route.options) route.options = {};
-      if (!route.options.bind) route.options.bind = context;
-      server.route(route);
+      route.options.bind = Object.assign(pluginContext, route.options.bind);
```

Comment:
If any `route.options.bind` is defined it will be added to the pluginContext and shared with all subsequent routes, so better not to mutate `pluginContext` and rather create a new object every time.
```
route.options.bind = Object.assign({}, pluginContext, route.options.bind);
```

Reply:
👍 Glad you're on this review. Wish you were here to prevent me from deleting those characters I had initially :)

Fixed(?) in https://github.com/elastic/kibana/commit/aef7ffb2608327014aa11034d8189749bff04a65